sstables/sstable_set.cc

```diff
@@ -227,21 +232,31 @@ dht::partition_range partitioned_sstable_set::to_partition_range(const dht::ring
 partitioned_sstable_set::partitioned_sstable_set(schema_ptr schema, lw_shared_ptr<sstable_list> all, bool use_level_metadata)
         : _schema(std::move(schema))
         , _all(std::move(all))
-        , _use_level_metadata(use_level_metadata) {
+        , _use_level_metadata(use_level_metadata)
+{
+    _sstable_generations.reserve(_all->size());
+    for (auto& sst : *_all) {
+        auto [it, inserted] = _sstable_generations.emplace(sst->generation());
+        if (!inserted) {
+            on_internal_error(sstlog, format("partitioned_sstable_set: could not ingest SSTable with generation {}", sst->generation()));
```

Comment:
Better make it explicit that the problem is with a duplicate sstable generation, and also mention what this generation is.

Reply:
OK, I'll try to rephrase it better.
BTW, the generation is already printed here.

Reply:
Clarified in v5